Sources/NIO/CircularBuffer.swift

```diff
@@ -24,6 +24,11 @@ public protocol AppendableCollection: Collection {
 /// will automatically expand if more elements than `initialRingCapacity` are stored, it's advantageous to prevent
 /// expansions from happening frequently. Expansions will always force an allocation and a copy to happen.
 public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
+    #if swift(>=4.2)
```

Comment:
Should this be added in multiple files? What about just one file with the `#if` and `#else`? That also makes it easier to find and clean up in the future.

Reply:
@BasThomas I don't like it either but I wanted to have the scope of this typealias to be as small as possible. The best would've been to just use `Swift.Range` but in Swift < 4.2 `Range` and `CountableRange` are distinct types therefore a change would've been SemVer major!